templates/routing-legacy/main.go

```diff
@@ -35,41 +39,149 @@ func solver(
+	if err != nil {
+		return FleetOutput{}, err
+	}
+
+	solutions, err := solver.Solve(ctx, options.Solve)
+	if err != nil {
+		return FleetOutput{}, err
 	}
 
-	solutions, err := solver.Solve(ctx, solveOptions)
+	// Last solution == 1, all solutions == 0
+	if runSolutions == 1 {
+		last := solutions.Last()
+		output, err := format(ctx, options.Solve.Duration, solver, ToFleetSolutionOutput, last)
+		if err != nil {
+			return output, err
+		}
+		return output, nil
+	}
+
+	solutionArray := make([]nextroute.Solution, 0)
+	for s := range solutions {
+		solutionArray = append(solutionArray, s)
+	}
+	output, err := format(ctx, options.Solve.Duration, solver, ToFleetSolutionOutput, solutionArray...)
+	if err != nil {
+		return output, err
+	}
+	return output, nil
+}
+
+func format(
+	ctx context.Context,
+	duration time.Duration,
+	progressioner alns.Progressioner,
+	toSolutionOutputFn func(nextroute.Solution) (any, error),
+	solutions ...nextroute.Solution,
+) (FleetOutput, error) {
+	mappedSolutions, err := MapWithError(solutions, toSolutionOutputFn)
 	if err != nil {
-		return runSchema.Output{}, err
+		return FleetOutput{}, err
+	}
+
+	output := FleetOutput{
+		Solutions: mappedSolutions,
+		Options: schema.Options{
+			Solver: &schema.SolverOptions{
+				Limits: &schema.Limits{
+					Duration: duration.String(),
+				},
+			},
+		},
+		Hop: struct {
+			Version string "json:\"version\""
+		}{
+			Version: sdk.VERSION,
+		},
+	}
+
+	startTime := time.Time{}
+	if start, ok := ctx.Value(run.Start).(time.Time); ok {
+		startTime = start
+	}
+
+	progressionValues := progressioner.Progression()
+
+	if len(progressionValues) == 0 {
+		return output, errors.New("no solution values or elapsed time values found")
+	}
+
+	seriesData := common.Map(
+		progressionValues,
+		func(progressionEntry alns.ProgressionEntry) statistics.DataPoint {
+			return statistics.DataPoint{
+				X: statistics.Float64(progressionEntry.ElapsedSeconds),
+				Y: statistics.Float64(progressionEntry.Value),
+			}
+		},
+	)
+
+	if len(output.Solutions) == 1 {
+		seriesData = seriesData[len(seriesData)-1:]
 	}
-	last := solutions.Last()
 
-	output := factory.Format(ctx, options, solver, last)
-	output.Statistics.Result.Custom = factory.DefaultCustomResultStatistics(last)
+	if len(output.Solutions) != len(seriesData) {
+		return output, errors.New("more or less solution values than solutions found")
+	}
+	for idx, data := range seriesData {
+		if _, ok := output.Solutions[idx].(FleetState); ok {
+			output.Statistics.Time.Start = startTime
+			output.Statistics.Value = int(data.Y)
+			output.Statistics.Time.ElapsedSeconds = float64(data.X)
+			output.Statistics.Time.Elapsed = time.Duration(data.X * statistics.Float64(time.Second)).String()
+		}
+	}
 
 	return output, nil
 }
+
+// MapWithError maps a slice of type T to a slice of type R using the function f.
+func MapWithError[T any, R any](v []T, f func(T) (R, error)) ([]R, error) {
```

Comment:
Does not have to be exported (like format)

Reply:
changed and moved to `helper.go`